### PR TITLE
feat: proxy profiles

### PR DESCRIPTION
**Pull request by kalinkrustev**
This is an overview of the changes:
- To simplify proxy deployment, some defaults settings are set, based on these assumptions:
  - We assume that scheme A of the proxy is the environment being deployed, so there is no need to explicitly configure it
  - As we are connecting the proxies to iac based deployments, we can derive some of the settings based on the domain name, so these settings have some defaults, based on the domain of the scheme B environment
- Added the ability for overriding values for the payment managers in `pm4ml-values-override.yaml`, which can be used in profiles or in custom-config
- Added the ability for overriding values for the proxies in `proxy-values-override.yaml`, which can be used in profiles or in custom-config
- Added the ability to configure the hub name by setting `hub_name` in `custom-config/mojaloop-vars.yaml`
- Hub name is by default set to the environment name prefixed with 'hub-'
- Clients are generated in Keycloak for each proxy
- Values are set in Tenancy vault for each proxy
- All proxies are added to the mcm DB via a seed
- Some fixes for the Redis operator

### CODE REVIEW

File: terraform/gitops/proxy-pm4ml/pm4ml.tf [hunk 1]
```diff
@@ -8,17 +8,23 @@ module "generate_pm4ml_files" {
...
     storage_class_name                              = var.storage_class_name
-    pm4ml_sync_wave                                 = var.pm4ml_sync_wave + index(keys(var.app_var_map), each.key)
+    pm4ml_sync_wave                                 = var.pm4ml_sync_wave
     external_load_balancer_dns                      = var.external_load_balancer_dns
     istio_internal_wildcard_gateway_name            = var.istio_internal_wildcard_gateway_name
     istio_internal_gateway_namespace                = var.istio_internal_gateway_namespace
     istio_external_wildcard_gateway_name            = var.istio_external_wildcard_gateway_name
     istio_external_gateway_namespace                = var.istio_external_gateway_namespace
     pm4ml_wildcard_gateway                          = each.value.pm4ml_ingress_internal_lb ? "internal" : "external"
-    proxy_id                                        = each.value.pm4ml_proxy_id
```
Comment (vijayg10) [inline terraform/gitops/proxy-pm4ml/pm4ml.tf:18]: I think there is a reason why we have pm4ml_proxy_id separately. There are differences in the allowed pattern for k8s resource names and DFSP IDs in the scheme. So we want them to be separate parameters.

File: terraform/gitops/pm4ml/pm4ml.tf [hunk 2]
```diff
@@ -59,7 +59,7 @@ module "generate_pm4ml_files" {
     pm4ml_external_switch_oidc_token_route          = each.value.pm4ml_external_switch_oidc_token_route
     pm4ml_external_switch_client_secret             = var.pm4ml_external_switch_client_secret
     pm4ml_external_switch_client_secret_key         = "token"
-    pm4ml_external_switch_client_secret_vault_key   = "${var.kv_path}/${var.cluster_name}/${each.key}/${each.value.pm4ml_external_switch_client_secret_vault_path}"
+    pm4ml_external_switch_client_secret_vault_key   = "${var.cluster_name}/${each.key}/${each.value.pm4ml_external_switch_client_secret_vault_path}"
```
Comment (vijayg10) [inline terraform/gitops/pm4ml/pm4ml.tf:62]: Why cluster_name is removed? I think we are accessing tenancy vault and there might be many clusters in that.

File: terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl [hunk 3]
```diff
@@ -40,35 +39,35 @@ inter-scheme-proxy-adapter:
...
-    MCM_SERVER_ENDPOINT: "https://${scheme_a_config.pm4ml_external_mcm_public_fqdn}/pm4mlapi"
+    HUB_IAM_PROVIDER_URL: "${oidc_origin_a}"
+    OIDC_TOKEN_ROUTE: "${oidc_path_a}"
+    MCM_SERVER_ENDPOINT: "https://${mcm_domain_a}/pm4mlapi"
     MCM_CLIENT_REFRESH_INTERVAL: 300
     PRIVATE_KEY_LENGTH: 2048
     PRIVATE_KEY_ALGORITHM: rsa
     AUTH_ENABLED: true
-    AUTH_CLIENT_ID: ${scheme_a_config.pm4ml_external_switch_client_id}
+    AUTH_CLIENT_ID: ${proxy_id}
```
Comment (vijayg10) [inline terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl:70]: Please see my other comment about client_id

Comment (vijayg10) [inline terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl:70]: Please see my other comment about client_id

File: terraform/gitops/proxy-pm4ml/pm4ml.tf (hunk 1 above)
Comment (kalinkrustev) [reply terraform/gitops/proxy-pm4ml/pm4ml.tf:18]: Also proxy ids are kind of for internal usage and we can choose proper/suitable names for them.

File: terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl [hunk 4]
```diff
@@ -40,35 +39,35 @@ inter-scheme-proxy-adapter:
...
-    PEER_ENDPOINT_A: "${scheme_a_config.pm4ml_external_switch_fqdn}"
-    PEER_ENDPOINT_B: "${scheme_b_config.pm4ml_external_switch_fqdn}"
-    OAUTH_TOKEN_ENDPOINT_A: "${scheme_a_config.pm4ml_external_switch_oidc_url}/${scheme_a_config.pm4ml_external_switch_oidc_token_route}"
-    OAUTH_CLIENT_KEY_A: "${scheme_a_config.pm4ml_external_switch_client_id}"
-    OAUTH_CLIENT_KEY_B: "${scheme_b_config.pm4ml_external_switch_client_id}"
+    PEER_ENDPOINT_A: "${peer_domain_a}"
+    PEER_ENDPOINT_B: "${peer_domain_b}"
+    OAUTH_TOKEN_ENDPOINT_A: "${oidc_origin_a}/${oidc_path_a}"
+    OAUTH_CLIENT_KEY_A: "${proxy_id}"
+    OAUTH_CLIENT_KEY_B: "${proxy_id}"
```
[the remark this answers is not in the document]
Comment (kalinkrustev) [reply terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl:46]: These client_ids are now onboarded by the Keycloak Realm Import. Even if manually onboarded, this is extra burden to set arbitrary client id and then go to 2 git repos and set these as parameters. Isn't it better to have some conventions and do less configuration?

File: terraform/gitops/proxy-pm4ml/pm4ml.tf (hunk 1 above)
Comment (vijayg10) [reply terraform/gitops/proxy-pm4ml/pm4ml.tf:18]: Actually we pass the same ID as DFSPID in the sdk scheme adapter which sends requests to switch with that ID. So its not internal. So lets imagine the switch provided a long DFSP ID and if we use that to deploy the instance, we may endup having some k8s resources exceed the 63 char limit. Which again we need to deal with some nameOverrides. So I think its unnecessary complication, let me know if it makes sense.

File: terraform/k8s/default-config/proxy-pm4ml-vars.yaml [hunk 5]
```diff
@@ -1,21 +1,12 @@
...
 pm4ml_scheme_b_config:
-  pm4ml_external_mcm_public_fqdn: mcm.dev.switch-b.mojaloop.live
-  pm4ml_external_switch_client_id: dfsp-jwt-b
-  pm4ml_external_switch_oidc_url: https://keycloak.dev.switch-b.mojaloop.live
-  pm4ml_external_switch_oidc_token_route: realms/dfsps/protocol/openid-connect/token
-  pm4ml_external_switch_client_secret_vault_path: "mcmdev_client_secret_b"
-  pm4ml_external_switch_fqdn: extapi.dev.switch-b.mojaloop.live
-pm4ml_proxy_id: proxypm4mltest
+  domain: buffer.${domain}
+  pm4ml_external_switch_client_secret_vault_path: mcm_client_secret_b
```
[the remark this answers is not in the document]
Comment (vijayg10) [reply terraform/k8s/default-config/proxy-pm4ml-vars.yaml:9]: Oh ok. If these variables are still available, then no issue for me.

File: terraform/gitops/proxy-pm4ml/pm4ml.tf [hunk 6]
```diff
@@ -43,8 +49,8 @@ module "generate_pm4ml_files" {
     pm4ml_external_switch_a_client_secret           = var.pm4ml_external_switch_a_client_secret
     pm4ml_external_switch_b_client_secret           = var.pm4ml_external_switch_b_client_secret
     pm4ml_external_switch_client_secret_key         = "token"
-    pm4ml_external_switch_a_client_secret_vault_key = "${var.kv_path}/${var.cluster_name}/${each.key}/${each.value.pm4ml_scheme_a_config.pm4ml_external_switch_client_secret_vault_path}"
-    pm4ml_external_switch_b_client_secret_vault_key = "${var.kv_path}/${var.cluster_name}/${each.key}/${each.value.pm4ml_scheme_b_config.pm4ml_external_switch_client_secret_vault_path}"
+    pm4ml_external_switch_a_client_secret_vault_key = "${var.cluster_name}/${each.key}/${each.value.pm4ml_scheme_a_config.pm4ml_external_switch_client_secret_vault_path}"
+    pm4ml_external_switch_b_client_secret_vault_key = "${var.cluster_name}/${each.key}/${each.value.pm4ml_scheme_b_config.pm4ml_external_switch_client_secret_vault_path}"
```
[the remark this answers is not in the document]
Comment (vijayg10) [reply terraform/gitops/proxy-pm4ml/pm4ml.tf:53]: Oh ok, please review this with @dfry also just incase.

File: terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (kalinkrustev) [reply terraform/gitops/generate-files/templates/proxy-pm4ml/values-proxy-pm4ml.yaml.tpl]: fixed in 40e38de11c3e7bcfe903f3469b41a0619522306c

File: terraform/gitops/proxy-pm4ml/pm4ml.tf (hunk 1 above)
Comment (kalinkrustev) [reply terraform/gitops/proxy-pm4ml/pm4ml.tf:18]: fixed in 40e38de11c3e7bcfe903f3469b41a0619522306c

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**